### PR TITLE
wallet_rpc_payments: implement multithreading

### PR DESCRIPTION
**Pull request by gdmojo**
Opening PR for early review. Will solve issue #7179

### CODE REVIEW

**Comment by moneromooo-monero** [comment]
I hate the reindent diff, but I guess it has to be. Seems OK, except hardcoding threads to 4 is not good, but easy to change.

**Comment by gdmojo** [comment]
Implemented user specified threads defaulting to 1.

File: src/simplewallet/simplewallet.cpp [hunk 1]
```diff
@@ -2360,6 +2360,25 @@ bool simple_wallet::start_mining_for_rpc(const std::vector<std::string> &args)
...
+  else
+  {
+    m_rpc_payment_threads = 1;
+  }
+
+  if (!ok)
+  {
+    PRINT_USAGE(USAGE_START_MINING_FOR_RPC);
+    return true;
+  }
```
Comment (moneromooo-monero) [inline src/simplewallet/simplewallet.cpp:2380]: This looks like if you run:
> start_mining_for_rpc 1
> start_mining_for_rpc 0

Then second one will fail as ok will be false, but m_rpc_payment_threads will still be set and will find its way into the call to search_for_rpc_payment later. Admittedly it's a "don't do that then" situation :)

**Comment by SamsungGalaxyPlayer** [comment]
Can the default be the max # of threads on the system? Is there an easy way to detect that?

**Comment by moneromooo-monero** [comment]
Yes, boost::thread::hardware_concurrency()

File: src/simplewallet/simplewallet.cpp (hunk 1 above)
Comment (gdmojo) [reply src/simplewallet/simplewallet.cpp:2379]: Well spotted.

**Comment by gdmojo** [comment]
Changed `m_rpc_payment_threads` so that value of `0` defaults to `boost::thread::hardware_concurrency` threads.

**Comment by mj-xmr** [comment]
> Changed `m_rpc_payment_threads` so that value of `0` defaults to `boost::thread::hardware_concurrency` threads.

Not super important, but in case you want to know, since C++11, there's also:

std::thread::hardware_concurrency

https://en.cppreference.com/w/cpp/thread/thread/hardware_concurrency

**Review by moneromooo-monero — APPROVED** [review]

**Pull request merged**